File: Week 2/problem_set_1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // how much change is owed in dollar
    float change;
    do
    {
        change = get_float("How much change is owed?\n");
    }
    while (change < 0);

    // convert dollors into cents
    int cents = round(change * 100);

    // only coins available are
    // quarters (25), dimes (10), nickels (5), and pennies (1)
    int quaters = floor(cents / 25);
    int dimes = floor((cents - quaters * 25) / 10);
    int nickels = floor((cents - quaters * 25 - dimes * 10) / 5);
    int pennies = cents - quaters * 25 - dimes * 10 - nickels * 5;
    // print the total number of coins
    printf("%i\n", quaters + dimes + nickels + pennies);
}

File: week_6/problem_set_6/cash/cash.py
from cs50 import get_float

# get user input as the change owed in dolloars
change = -1
while change < 0:
    change = get_float("Change owed: ")

# number of cents owed calculated from change
cents = change * 100

# calculate the number of coins
quarters = cents // 25
cents -= quarters * 25

dimes = cents // 10
cents -= dimes * 10

nickels = cents // 5
cents -= nickels * 5

pennies = cents

print(int(quarters + dimes + nickels + pennies))